x-pack/plugins/observability_solution/infra/server/plugin.ts

```diff
@@ -212,12 +212,22 @@ export class InfraServerPlugin
       metrics: new InfraMetricsDomain(new KibanaMetricsAdapter(framework)),
     };
 
+    const libsPlugins = mapValues(plugins, (value, key) => {
```

Comment:
I will add a TODO here.

`libs` currently receives individual plugins that some downstream service might depend on. by passing an object with all plugins we take the burden away of always having to change the `libs` type and adding a new object to it. That's what APM and profiling do